src/qml/editorwidgets/RelationReference.qml

```diff
@@ -8,21 +8,52 @@ import org.qgis 1.0
 import Theme 1.0
 import ".."
 
+Item {
+  anchors { left: parent.left; right: parent.right; }
 
-RelationCombobox {
-  id: relationReference
-  enabled: isEnabled
+  property bool showOpenFormButton: config['ShowOpenFormButton'] === undefined || config['ShowOpenFormButton'] === true
 
   signal valueChanged(var value, bool isNull)
 
-  property var _relation: qgisProject.relationManager.relation(config['Relation'])
+  RelationCombobox {
+    id: relationReference
+    anchors { left: parent.left; right: parent.right; rightMargin: showOpenFormButton ? 24 : 0 }
+    enabled: isEnabled
 
-  FeatureListModel {
-    id: featureListModel
+    signal valueChanged(var value, bool isNull)
 
-    currentLayer: qgisProject.relationManager.relation(config['Relation']).referencedLayer
-    keyField: qgisProject.relationManager.relation(config['Relation']).resolveReferencedField(field.name)
-    addNull: config['AllowNULL']
-    orderByValue: config['OrderByValue']
+    property var _relation: qgisProject.relationManager.relation(config['Relation'])
+
+    FeatureListModel {
+      id: featureListModel
+
+      currentLayer: qgisProject.relationManager.relation(config['Relation']).referencedLayer
+      keyField: qgisProject.relationManager.relation(config['Relation']).resolveReferencedField(field.name)
+      addNull: config['AllowNULL']
+      orderByValue: config['OrderByValue']
+    }
+
+    onValueChanged: parent.valueChanged(value,isNull)
+  }
+
+  Image {
+    id: viewButton
+    anchors { right: parent.right; top: relationReference.top; topMargin: relationReference.childrenRect.height / 2 - 9 }
+    source: Theme.getThemeIcon("ic_view_green_48dp")
+    enabled: showOpenFormButton && relationReference.currentKeyValue !== undefined
```

Comment:
Is it enabled when `NULL` is selected?

Reply:
Just made a test with `allowNull` - it seems like it's displayed on a `NULL` value stored.

And when I add a new child when `allowNull` is active, it selects `NULL` (instead of the newly added feature). I didn't dig in it still. Tell me when you cannot reproduce.

Reply:
This allowNull thing works OK for me:
![Peek 2020-11-02 18-34](https://user-images.githubusercontent.com/1728657/97863780-215d3f80-1d3a-11eb-937f-5a7dce188cf6.gif)

Reply:
Save it. And open form again.
And try to add a new entry. It seems to select NULL then.
But maybe it's only my project.

Reply:
Can you make a demo GIF?

Reply:
![videotogif_2020 11 02_13 15 55](https://user-images.githubusercontent.com/28384354/97867831-ded23d80-1d0e-11eb-9dab-2c752cc68cb8.gif)
![Uploading videotogif_2020.11.02_13.14.59.gif…]()

Reply:
![videotogif_2020 11 02_13 14 59](https://user-images.githubusercontent.com/28384354/97867978-28228d00-1d0f-11eb-976c-a3286a43ac51.gif)

Reply:
@signedav , OK, fixed.

Reply:
Visibility is fixed, needed check for empty string makes sense 👍 
But after creating a parent with + it still  does not select the created feature but NULL instead. (Since I'm only on mobile I only tested the apk and didn't checked out the code.)

Reply:
@signedav , that's a bug that predates this work, I'll fix in a different PR if you don't mind.